server/controllers/BookController.js

```diff
@@ -75,6 +77,42 @@ class BookController {
       return responseMessage(response, 500, { message: error.message });
     }
   }
+
+  /**
+ * Add a new book
+ * @name addBook
+ * @param {Object} request
+ * @param {Object} response
+ * @returns {JSON} object
+ */
+  static async getBorrowingStats(request, response) {
+    try {
+      const user = JSON.parse(JSON.stringify(request.userData));
+      const { id } = user;
+      let filter = { userId: id };
+      const { roleName } = user.UserRoles[0].Role;
+      if (roleName === 'admin' || roleName === 'superadmin') filter = {};
+      const { limit, page } = request.query;
+      const { count } = await LendingHistory.findAndCountAll({ where: filter });
+      const totalPages = Math.ceil(count / (limit || 10));
+      if ((page || 1) > totalPages) return responseMessage(response, 400, { status: 'failure', message: 'no book(s) found on this page' });
```

Comment:
When I tested this route using a user with no lending history, the response on this line was sent back. It is not a proper message for when a user doesn't have a lending history. Use a separate response message for that

Reply:
Corrected